widgetssdk/src/main/java/com/glia/widgets/view/head/ActivityWatcherForChatHead.kt

```diff
@@ -38,15 +51,86 @@ internal class ActivityWatcherForChatHead(
 
     override fun onActivityPaused(activity: Activity) {
         controller.onActivityPaused()
+        removeChatHeadViewIfPresent()
         resumedActivity.clear()
     }
 
+    private fun createBubble(activity: Activity) {
+        val bubble = ChatHeadLayout(activity.baseContext)
+        bubble.layoutParams = ConstraintLayout.LayoutParams(
+            ViewGroup.LayoutParams.MATCH_PARENT,
+            ViewGroup.LayoutParams.MATCH_PARENT
+        )
+        bubble.setNavigationCallback(
+            object : ChatHeadLayout.NavigationCallback {
+                override fun onNavigateToEndScreenSharing() {
+                    navigateToEndScreenSharing(resumedActivity.get())
+                }
+
+                override fun onNavigateToChat() {
+                    navigateToChat(resumedActivity.get())
+                }
+
+                override fun onNavigateToCall() {
+                    navigateToCall(resumedActivity.get())
+                }
+            }
+        )
+        bubble.visibility = View.VISIBLE
+        this.bubble = WeakReference(bubble)
+    }
+
+    override fun addChatHeadViewIfAbsent() {
+        val viewName = fetchGliaOrRootView()?.javaClass?.simpleName
+        if (hasGliaView(resumedActivity.get()) || !controller.shouldShowBubble(viewName)) return
+        (fetchGliaOrRootView() as? ViewGroup)?.let { viewGroup ->
+            if (!viewGroup.hasChildOfType(ChatHeadLayout::class.java))
+                try {
+                    resumedActivity.get()?.let { activity ->
+                        createBubble(activity)
+                        Logger.d(TAG, "Adding application-only bubble")
+                        resumedActivity.get()?.runOnUiThread { viewGroup.addView(bubble.get()) }
+                    }
+                } catch (e: IllegalStateException) {
+                    Log.d(TAG, "Cannot add bubble: $e")
+                }
+        }
```

Comment:
NIT: no problem with logic, but can you please make it a bit more readable?
Not perfect but I think it's a bit easier for the eye, example:
```suggestion
        val viewName = fetchGliaOrRootView()?.javaClass?.simpleName
        val activity = resumedActivity.get() ?: return
        if (hasGliaView(activity) || !controller.shouldShowBubble(viewName)) return
        val viewGroup = (fetchGliaOrRootView() as? ViewGroup) ?: return
        if (!viewGroup.hasChildOfType(ChatHeadLayout::class.java)) {
            try {
                createBubble(activity)
                Logger.d(TAG, "Adding application-only bubble")
                activity.runOnUiThread { viewGroup.addView(bubble.get()) }
            } catch (e: IllegalStateException) {
                Log.d(TAG, "Cannot add bubble: $e")
            }
        }
```

Reply:
Yes, agree. Refactored. f82ece80e3aa6fe3ca03d67bc53b38c5ae726f7e